### DISCUSSION TITLE
Training NN - handwritten notes

### DISCUSSION THREAD

**Discussion by b-pobiedzinska** [Audiveris/audiveris#729] [Q&A]
I started working with audiveris, and was wondering whether with the current framework it is even possible for the program to recognize handwritten notes? I know that it is not supported, but I was wondering if by training the neural network with such examples, or maybe changing its architecture to a more complex one it would be doable?

**Comment by hbitteur** [comment]
@b-pobiedzinska 

The current Audiveris approach is based on a mix of:
- _ad hoc_ methods to retrieve key components like staves, barlines, stems, beams, slurs, ...
- template matching for note heads
- neural network for other fixed-size items

So, training the neural network would impact only the 3rd point above.

A more global approach can be envisioned. See for example [our wiki article](https://github.com/Audiveris/audiveris/wiki/6x-Presentation) on the 6.0 prototype.
Last year, I tried a patch-classifier, able to recognize the item, if any, centered at any given point in the input image. It worked, but was so slow that processing each and every pixel of the input image would take several days!
This year, instead of a sliding window, I intend to try an SSD (single-shot detection) approach. Stay tuned... :-)
